binder/postBuild

```diff
@@ -1 +1,46 @@
-jupyter labextension install jupyterlab-js-logs@0.1.1 --debug
+#!/usr/bin/env python3
+""" perform a development install of jupyterlab-js-logs
+
+    On Binder, this will run _after_ the environment has been fully created from
+    the environment.yml in this directory.
+
+    This script should also run locally on Linux/MacOS/Windows:
+
+        python3 binder/postBuild
+"""
+import subprocess
+import sys
+from pathlib import Path
+
+
+ROOT = Path.cwd()
+
+def _(*args, **kwargs):
+    """ Run a command, echoing the args
+
+        fails hard if something goes wrong
+    """
+    print("\n\t", " ".join(args), "\n")
+    return_code = subprocess.call(args, **kwargs)
+    if return_code != 0:
+        print("\nERROR", return_code, " ".join(args))
+        sys.exit(return_code)
+
+# verify the environment is self-consistent before even starting
+_(sys.executable, "-m", "pip", "check")
+
+# install the labextension
+_(sys.executable, "-m", "pip", "install", "-e", ".")
```

Comment:
Maybe removing "-e" will fix the binder? It seems that with "-e" it doesn't install the labextension for me locally, I guess it's due to the new jupyter-packaging version? `pip install .` does install the labextension for me
.
```suggestion
_(sys.executable, "-m", "pip", "install", ".")
```

Reply:
Normally it should indeed be possible to `pip install -e .` the extension just like before, and have it installed.

Reply:
Just checked locally with a new extension from the cookiecutter with:

```bash
cookiecutter https://github.com/jupyterlab/extension-cookiecutter-ts
cd myextension
python -m pip install -e .
```

And the extension is indeed not listed:

```bash
$ jupyter labextension list
JupyterLab v3.0.14
```

We should open an issue in https://github.com/jupyterlab/extension-cookiecutter-ts to track this.

Reply:
Before `jupyter-packaging`, we used to not have `nbextensions` installed when running `pip install -e .`, meaning we didn't have to pass the `--overwrite` option when installing a symlink of the extension. It was also a fine workflow.

I guess having `pip install -e .` install the extension is more convenient for people who only want to touch the Python code, less convenient for people who want to debug the JS/TS code. For labextensions, the Python being only one function it might make sense to **not** install  the extension, and let the developer run `jupyter labextension develop .` (without `--overwrite`)

Reply:
The cookiecutter was already using `jupyter-packaging` before, but `pip install -e .` would also install the extension assets (build in prod / minified mode).

Now it's mostly a question of whether `pip install -e .` should still build and install the lab extension under `$PREFIX/share/jupyter/labextensions` like before, or leave it to the developer to use the `jupyter labextension develop` command as an extra step like it seems to be the case now.

Reply:
But otherwise yes, updating the postBuild step on Binder to use `pip install .` sounds good :+1:

Reply:
> The cookiecutter was already using jupyter-packaging before, but pip install -e .

Sure, but there was a time when jupyter-packaging didn't exist, and we were not installing the extension for dev install (at least for widget extensions).

> Now it's mostly a question of whether pip install -e . should still build and install the lab extension under $PREFIX/share/jupyter/labextensions like before, or leave it to the developer to use the jupyter labextension develop command as an extra step like it seems to be the case now.

Yeah. I would personnaly be 👍🏼 for not installing the extension for a dev install.